src/workspaces/useWorkspaces.ts

```diff
@@ -0,0 +1,45 @@
+import { FieldsArg, workspaceProvider } from 'src/libs/ajax/workspaces/providers/WorkspaceProvider';
+import { useCancellation } from 'src/libs/react-utils';
+import { workspacesStore } from 'src/libs/state';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+import { useWorkspacesData, UseWorkspacesStateResult } from 'src/workspaces/useWorkspacesData';
+
+const defaultFieldsArgs: FieldsArg = [
+  'accessLevel',
+  'public',
+  'workspace',
+  'workspace.state',
+  'workspace.attributes.description',
+  'workspace.attributes.tag:tags',
+  'workspace.workspaceVersion',
+];
+
+/**
+ * A composed version of useWorkspacesData hook that adds Terra-UI specific data-access and concerns.
+ * Honors expected hook return contract.
+ * @param fieldsArg
+ * @param stringAttributeMaxLength
+ */
+export const useWorkspaces = (fieldsArg?: FieldsArg, stringAttributeMaxLength?: number): UseWorkspacesStateResult => {
```

Comment:
AoU would compose a similar version of useWorkspaces for use in composing Envoronments component props in their version of EnvironmentsPage.ts